src/wallet/rpcwallet.cpp

```diff
@@ -768,8 +770,10 @@ UniValue getbalance(const JSONRPCRequest& request)
 
     LOCK2(cs_main, pwallet->cs_wallet);
 
-    if (request.params.size() == 0)
-        return  ValueFromAmount(pwallet->GetBalance());
+    bool ignoreUnspendable = request.params[3].isNull() || !request.params[3].get_bool();
+    if ((request.params[0].isNull() || (request.params[0].get_str() == "" && !ignoreUnspendable)) && request.params[1].isNull() && request.params[2].isNull()) {
```

Comment:
This is confusing for me. `include_unspendable` only has effect if:

- `account` is not specified; and
- `minconf` is not specified; and
- `include_watchonly` is not specified

Additionally, if `account` is specified as `""` (ie balance not associated with any account), the returned balance will use `GetBalance()` if `include_unspendable` is not set, but `GetLegacyBalance()` if it is set.

Can you explain the reasoning here?